engine/src/main/java/org/hibernate/validator/internal/properties/Constrainable.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.properties;
+
+import java.lang.reflect.Type;
+
+import org.hibernate.validator.internal.metadata.raw.ConstrainedElement.ConstrainedElementKind;
+
+/**
+ * @author Marko Bekhta
+ */
+public interface Constrainable {
+
+	String getName();
+
+	Class<?> getDeclaringClass();
```

Comment:
Doesn't this break the "abstraction"? I.e. what to return for a JSON type?

Reply:
Mmmh, yes it does. I think @marko-bekhta had in mind to provide a `PropertyHolder` class here or something similar.

The class is used in some portions of the code where it's not really easy to abstract from it.

Let's see how it goes with the JSON PR and act accordingly.

I created https://hibernate.atlassian.net/browse/HV-1632 so that we have another try at it once we got the JSON support into shape.